PICARD-2715: Show plugins with updates and add button to open plugin options window

rdswift opened the pull request:
# Summary

* This is a…
  * [ ] Bug fix
  * [x] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:  Adds list of plugin updates available and a button to open the Options -> Plugins screen directly from the plugin update notification dialog

# Problem

The plugin updates available notification dialog did not provide any details or allow easy updating.

* JIRA ticket (_optional_): PICARD-2715

# Solution

Add a list of plugin updates available and a button to open the Options -> Plugins screen directly from the plugin update notification dialog.  Examples:

![image](https://github.com/metabrainz/picard/assets/16990904/d523803d-506f-47fe-9b09-c64fe15a3810)

![image](https://github.com/metabrainz/picard/assets/16990904/cc789a82-9648-4475-8a37-35d2eb32ea88)

Note that the displayed list is limited to a maximum of 3 plugins, with a note indicating if there are more.

# Action

None.

phw reviewed (dismissed):
Nice and clean. I think this makes the update check really more useful.

rdswift commented:
Do we still want to default this plugin check to disabled?

Should I add a checkbox to the dialog to disable the checks, similarly to the new user / first run / save warning dialogs?

phw commented:
I think we should keep it disabled for now. We already introduced multiple new popups. I think it is better to not show another one unasked on a minor update.

rdswift commented:
I added to option to disable further checks, and refactored a bit.  The dialog now looks like:

![image](https://github.com/metabrainz/picard/assets/16990904/828068d7-c5ac-42d7-a236-039155237aae)

rdswift commented on picard/pluginmanager.py:
```diff
@@ -449,26 +482,55 @@ def enabled(self, name):
+                msg = PluginUpdatesDialog(parent, f'{header}<ul>{plugin_list}</ul>{extra_plugins}{footer}')
+
+                show_options_page, perform_check = msg.show()
+                if parent:
+                    config = get_config()
+                    config.setting['check_for_plugin_updates'] = perform_check
+
+                    if show_options_page:
+                        parent.show_plugins_options_page()
```
Should this be changed to:
```suggestion
                if parent:
                config = get_config()
                config.setting['check_for_plugin_updates'] = perform_check

                if parent and show_options_page:
                    parent.show_plugins_options_page()
```

rdswift replied:
Changes tested and implemented.

zas replied:
Yes I think so. The setting doesn't depend on `parent`.
